### PR TITLE
Add changelog for v3.1.0

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Preparation for v3.1.0 / Kubernetes 1.24.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by pohly** [comment]
/assign @xing-yang 

For approval.

**Comment by pohly** [comment]
/hold

Let's include https://github.com/kubernetes-csi/external-provisioner/pull/688 in the next release.

**Comment by pohly** [comment]
/hold cancel

All pending PRs are merged and this one was updated, we can release.

File: CHANGELOG/CHANGELOG-3.1.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Release notes for v3.1.0
+
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v3.0.0
+
+## Changes by Kind
+
+### Feature
+ - Introduces `HonorPvReclaim` feature gate that ensures PV honors the Reclaim policy irrespective of deletion order. ([#679](https://github.com/kubernetes-csi/external-provisioner/pull/679), [@deepakkinni](https://github.com/deepakkinni))
```
Comment (xing-yang) [inline CHANGELOG/CHANGELOG-3.1.md:10]: It is "HonorPVReclaimPolicy".

**Comment by humblec** [comment]
lgtm.. Thanks !

File: CHANGELOG/CHANGELOG-3.1.md (hunk 1 above)
Comment (humblec) [reply CHANGELOG/CHANGELOG-3.1.md:10]: it is actually wrongly mentioned in the PR release note itself :(..

Comment (pohly) [reply CHANGELOG/CHANGELOG-3.1.md]: I've fixed it in this PR.

**Pull request merged**